tests/sles4sap/sap_deployment_automation_framework/publiccloud_lib_compatibility_layer.pm

```diff
@@ -0,0 +1,87 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Prepares compatibility layer for using `lib/publiccloud/*` library with SDAF deployment
+
+use parent 'sles4sap::sap_deployment_automation_framework::basetest';
+
+use warnings;
+use strict;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use sles4sap::console_redirection qw(connect_target_to_serial disconnect_target_from_serial);
+use sles4sap::sap_deployment_automation_framework::inventory_tools qw(read_inventory_file sdaf_create_instances);
+use sles4sap::sap_deployment_automation_framework::naming_conventions
+  qw(get_sdaf_inventory_path convert_region_to_short get_sdaf_config_path get_workload_vnet_code);
+
+sub run {
+    my ($self, $run_args) = @_;
+    select_serial_terminal;
+    my $workload_vnet_code = get_workload_vnet_code();
+    my $sap_sid = get_required_var('SAP_SID');
+    my $sdaf_region_short = convert_region_to_short(get_required_var('PUBLIC_CLOUD_REGION'));
+    my $sdaf_env_code = get_required_var('SDAF_ENV_CODE');
+
+    my $inventory_path = get_sdaf_inventory_path(
+        env_code => $sdaf_env_code,
+        sdaf_region_code => $sdaf_region_short,
+        vnet_code => $workload_vnet_code,
+        sap_sid => $sap_sid
+    );
+
+    my $sut_ssh_private_key = get_sdaf_config_path(
+        deployment_type => 'sap_system',
+        env_code => $sdaf_env_code,
+        sdaf_region_code => $sdaf_region_short,
+        vnet_code => $workload_vnet_code,
+        sap_sid => $sap_sid) . '/sshkey';
+
+    # Redirect serial to deployer VM. Deployer VM takes same role as worker VM.
+    # Normally PC test runs under root user
+    connect_target_to_serial(switch_root => 'yes');
+
+    # PC lib uses 'bernhard' user by default - change global variable to override this default value
+    $testapi::username = "root";
+
+    my $inventory_data = read_inventory_file($inventory_path);
+
+    # Copy SUT ssh key to root user home dir which is what PC library is using by default.
+    assert_script_run("cp -Rp $sut_ssh_private_key /root/.ssh/id_rsa");
+    # Copy known_hosts file from azureadm user
+    assert_script_run("cp /home/azureadm/.ssh/known_hosts /root/.ssh/");
```

Comment:
Is there no way on PC side to say "Run all like SOMEUSER" natively? Is it a feature we can request (variable like PUBLICCLOUD_DEFAUL_USER to control it, or something like this)

Reply:
few like before https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20749/files#diff-4b3638dc52b7f82f4782f3ae797a3b328faaf1bf358bfbf586de2177df7453eaR45 there's a comment states `PC lib uses 'bernhard' user by defaul`  and here `to root user home dir which is what PC library is using by default`

Reply:
We could do it, but probably in another PR.
I got yesterday feedback that we will take the on prem test writing approach for SDAF and redirect console straight to SUT.

Reply:
Cool, so we can use assert_script_run liked api directly.